status/statusgo_backend_new/stickers.nim

```diff
@@ -0,0 +1,74 @@
+import json, stint, web3/[ethtypes, conversions], strutils, tables, sequtils
+import
+  ../types/[setting],
+  ../transactions,
+  ../utils as status_utils,
+  ../statusgo_backend/wallet,
+  ../eth/contracts as eth_contracts,
+  ../eth/stickers as eth_stickers,
+  ../eth/methods as eth_methods,
+  ../statusgo_backend/settings as status_settings,
+  ../types/[sticker, transaction, rpc_response, network_type, network]
+
+proc buildTransaction(
+    packId: Uint256,
+    address: Address,
+    price: Uint256,
+    approveAndCall: var ApproveAndCall[100],
+    sntContract: var Erc20Contract,
+    gas = "",
+    gasPrice = "",
+    isEIP1559Enabled = false,
+    maxPriorityFeePerGas = "",
+    maxFeePerGas = ""
+    ): TransactionData =
+  let network = status_settings.getCurrentNetwork().toNetwork()
+  sntContract = eth_contracts.findErc20Contract(network.chainId, network.sntSymbol())
+  let
+    stickerMktContract = eth_contracts.findContract(network.chainId, "sticker-market")
+    buyToken = BuyToken(packId: packId, address: address, price: price)
+    buyTxAbiEncoded = stickerMktContract.methods["buyToken"].encodeAbi(buyToken)
+  approveAndCall = ApproveAndCall[100](to: stickerMktContract.address, value: price, data: DynamicBytes[100].fromHex(buyTxAbiEncoded))
+  transactions.buildTokenTransaction(address, sntContract.address, gas, gasPrice, isEIP1559Enabled, maxPriorityFeePerGas, maxFeePerGas)
+
+proc getInstalledStickerPacks*(): JsonNode =
+  status_settings.getSetting[string](Setting.Stickers_PacksInstalled, "{}").parseJson
+
+proc getPurchasedStickerPacks*(address: Address): seq[int] =
+  let
+    network = status_settings.getCurrentNetwork().toNetwork()
+    balance = eth_stickers.getBalance(network.chainId, address)
+    tokenIds = toSeq[0..<balance].mapIt(eth_stickers.tokenOfOwnerByIndex(network.chainId, address, it.u256))
+    purchasedPackIds = tokenIds.mapIt(eth_stickers.getPackIdFromTokenId(network.chainId, it.u256))
+
+  result = purchasedPackIds
+
+proc getRecentStickers*(): seq[Sticker] =
+  eth_stickers.getRecentStickers()
+
+proc buyPack*(packId: int, address, price, gas, gasPrice: string, isEIP1559Enabled: bool, maxPriorityFeePerGas: string, maxFeePerGas: string, password: string, success: var bool): string =
+  var
+    sntContract: Erc20Contract
+    approveAndCall: ApproveAndCall[100]
+    tx = buildTransaction(
+      packId.u256,
+      status_utils.parseAddress(address),
+      status_utils.eth2Wei(parseFloat(price), 18), # SNT
+      approveAndCall,
+      sntContract,
+      gas,
+      gasPrice,
+      isEIP1559Enabled,
+      maxPriorityFeePerGas,
+      maxFeePerGas
+    )
+
+  result = sntContract.methods["approveAndCall"].send(tx, approveAndCall, password, success)
+  if success:
+    trackPendingTransaction(result, address, $sntContract.address, PendingTransactionType.BuyStickerPack, $packId)
+
+proc saveRecentStickers*(stickers: seq[string]) =
```

Comment:
@jrainville all methods from the status-lib should have the same return type `RpcResponse[JsonNode] {.raises: [Exception].}` regardless we're using that response or not, `status-lib` doesn't care about that, the decision about that will be made in appropriate service on the `status-desktop` side.